rideshare_worksheet.rb

```diff
@@ -0,0 +1,43 @@
+rides_by_driver =
+  [
+    [{:driver_id => "DR0001", :rider => "RD0003", :date => "3rd Feb 2016", :cost => 10, :rating => 3},
+     {:driver_id => "DR0001", :rider => "RD0015", :date => "3rd Feb 2016", :cost => 30, :rating => 4},
+     {:driver_id => "DR0001", :rider => "RD0003", :date => "5th Feb 2016", :cost => 45, :rating => 2}],
+
+    [{:driver_id => "DR0002", :rider => "RD0073", :date => "3rd Feb 2016", :cost => 25, :rating => 5},
+     {:driver_id => "DR0002", :rider => "RD0013", :date => "4th Feb 2016", :cost => 15, :rating => 1},
+     {:driver_id => "DR0002", :rider => "RD0066", :date => "5th Feb 2016", :cost => 35, :rating => 3}],
+
+    [{:driver_id => "DR0003", :rider => "RD0066", :date => "4th Feb 2016", :cost => 5, :rating => 5},
+     {:driver_id => "DR0003", :rider => "RD0003", :date => "5th Feb 2016", :cost => 50, :rating => 2}],
+
+    [{:driver_id => "DR0004", :rider => "RD0022", :date => "3rd Feb 2016", :cost => 5, :rating => 5},
+     {:driver_id => "DR0004", :rider => "RD0022", :date => "4th Feb 2016", :cost => 10, :rating => 4},
+     {:driver_id => "DR0004", :rider => "RD0073", :date => "5th Feb 2016", :cost => 20, :rating => 5}],
+  ]
+
+puts "~~NUMBER OF RIDES~~"
+rides_by_driver.each do |driver|
+  puts "#{driver[0][:driver_id]} : #{driver.count { |ride| ride[:driver_id] == driver[0][:driver_id] }}"
+end
```

Comment:
In general, it's considered bad form to do work inside a string interpolation. There's simply too much going on, and it's very hard to read. Instead, I would break this out across multiple lines and use some intermediate variables:

```ruby
driver_id = driver[0][:driver_id]
ride_count = driver.count do |ride|
  ride[:driver_id] == driver[0][:driver_id]
end
puts "#{driver_id} : #{ride_count}"
```

Reply:
I'm also not convinced by the variable name `driver`. Since the iteration variable contains a list of rides, something like `rides` might be a better choice.